src/Chainweb/Miner/Core.hs

```diff
@@ -12,32 +14,61 @@
 -- Stability: experimental
 
 module Chainweb.Miner.Core
-  ( usePowHash
+  ( HeaderBytes(..)
+  , TargetBytes(..)
+  , Bites(..), bites, unbites
+  , usePowHash
   , mine
   ) where
 
 import Crypto.Hash.Algorithms (SHA512t_256)
 import Crypto.Hash.IO
 
+import Data.Bifunctor (bimap)
 import qualified Data.ByteArray as BA
-import Data.Bytes.Put (runPutS)
 import qualified Data.ByteString as B
 import Data.Proxy (Proxy(..))
+import Data.Tuple.Strict (T2(..))
 import Data.Word (Word64, Word8)
 
 import Foreign.Marshal.Alloc (allocaBytes)
 import Foreign.Ptr (Ptr, castPtr)
 import Foreign.Storable (peekElemOff, poke)
 
+import Servant.API
+
 -- internal modules
 
-import Chainweb.BlockHeader
-import Chainweb.Difficulty (HashTarget, encodeHashTarget)
-import Chainweb.Utils (int, runGet)
+import Chainweb.BlockHeader (Nonce(..))
 import Chainweb.Version (ChainwebVersion(..))
 
 ---
 
+-- | Encoding of @HashTarget + BlockHeader@.
+newtype Bites = Bites { _bites :: B.ByteString }
+    deriving newtype (MimeRender OctetStream, MimeUnrender OctetStream)
```

Comment:
Perhaps a less stupid name?

Reply:
It's charming, but a little confusing a mnemonic 😄

Reply:
Changed.